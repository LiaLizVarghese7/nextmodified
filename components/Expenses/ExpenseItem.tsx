import ExpenseDate from "./ExpenseDate";
// import Card from "../UI/Card";
// import "./ExpenseItem.css";
import React, { ChangeEvent, useState } from "react";
import axios from "axios";

interface ExpensesItemType{
  id:number;
  title: string;
  amount: number;
  date: { 
    month: string;
     day: string;
     year: number;
  };
}

const ExpenseItem = (props: ExpensesItemType) => {
  const [title, setTitle] = useState(props.title);
  const [enteredTitle, setEnteredTitle] = useState('');
  const [message, setMessage] = useState('Enter updated title');
  const config = {
    headers: {
      "Access-Control-Allow-Origin": "*",
      "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS"
    }
  };
  let id = `${props.id}`;
  const url = `http://localhost:9000/products/`+ id ;
  const clickHandler = () => {
    console.log(title);
    if(enteredTitle!== ''){
    setTitle(enteredTitle);
    setMessage('Enter updated title');
    axios.patch(url, {title: enteredTitle}, config)
      .then(
        (response) => {
          console.log(response);
          console.log(response.status === 200, 'product updated');
        },
        (error) => {
          console.log(error);
        }
      )
      .catch((err) => console.log(err));
    }
    else{
      setMessage('Fill out this field to update');
    }
    setEnteredTitle('');
  };
  const titleChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
    
    setEnteredTitle(event.target.value);
  }
  console.log(props);
  return (
    // <Card className="flex justify-between items-center shadow-sm p-2 my-4 mx-0 rounded-xl bg-slate-300">
    //   <div className= "flex gap-4 my-2">
    //     <ExpenseDate date={props.date} />
    //     <h2 className="text-white text-base flex-auto mx-4 my-4">{title}</h2>
    //   </div>
    //   <div className="flex flex-row gap-4 items-end justify-start flex-auto">
        
        

    //     <div className= "flex flex-row flex-wrap gap-4 mb-2">
    //     <div className="text-base font-bold text-white bg-violet-800 border-solid border border-white p-2 rounded-xl">${props.amount}</div>
    //         <input className="p-2 rounded-md border border-solid border-gray-200" type="text" value= { enteredTitle } onChange={titleChangeHandler} />
          
    //       <button className="text-base font-bold text-white bg-violet-800 border-solid border border-white p-2 rounded-xl" onClick={clickHandler}>Change Title</button>
    //     </div>
    //   </div>
      
    // </Card>

    <div className="flex justify-between items-center shadow-sm p-2 my-4 mx-2 rounded-xl bg-slate-300">
      <div className="w-full sm:w-1/8 flex gap-2 items-end justify-start">
        <div className="w-full sm:w-1/2">
          <ExpenseDate date={props.date} />
        </div>
        <div className="w-full sm:w-1/2">
          <h2 className="text-black font-bold text-base flex-auto mx-4 mb-7">{title}</h2>
        </div>
      </div>
      {/* <div className="flex flex-col gap-5 items-end justify-start flex-auto"> */}
        <div className="w-full sm:w-7/8 flex flex-wrap gap-3 mb-2">
        <div className="w-full w-full md:w-1/4 text-base font-bold text-white bg-violet-800 border-solid border border-white p-2 rounded-xl">${props.amount}</div>
        <input className="w-full sm:w-1/3 md:w-1/4 p-2 rounded-md border border-solid border-gray-200" type="text" placeholder={message} value= { enteredTitle } required onChange={titleChangeHandler} />
        <button className="w-full sm:w-1/3 md:w-1/4 text-base font-bold text-white bg-violet-800 border-solid border border-white p-2 rounded-xl" onClick={clickHandler}>Change Title</button>
      </div>
      
    </div>
  );
};

export default ExpenseItem;
