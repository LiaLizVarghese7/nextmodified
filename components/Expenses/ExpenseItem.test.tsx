import { render } from "@testing-library/react";
import ExpenseItem from './ExpenseItem'
describe('Expense item date test', () => {
    it('should render UI correctly', () => {
        const mockItem ={
        id: 7, 
        title: 'Chocolate', 
        amount: 133, 
        date: {
            day:"18",
            month: "April",
            year: 2019}
        }
        expect(render(<ExpenseItem {...mockItem}/>)).toMatchSnapshot();
    });
});