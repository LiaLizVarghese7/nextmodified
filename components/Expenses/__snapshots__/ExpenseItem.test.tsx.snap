// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense item date test should render UI correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-red-300 w-20 h-20 border-2 rounded-xl border-white border-solid text-center justify-center"
      >
        <div
          class="text-xs font-bold"
        >
          April
        </div>
        <div
          class="text-xl font-bold"
        >
          18
        </div>
        <div
          class="text-2xl font-bold"
        >
          2019
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-red-300 w-20 h-20 border-2 rounded-xl border-white border-solid text-center justify-center"
    >
      <div
        class="text-xs font-bold"
      >
        April
      </div>
      <div
        class="text-xl font-bold"
      >
        18
      </div>
      <div
        class="text-2xl font-bold"
      >
        2019
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
